### PR TITLE
admin plugin support for additional guice modules

### PR DESCRIPTION
**Pull request by amit-git**
1. AdminPage plugin can provide a list of guice bindings that a plugin author can create. The modules will be included in the Guice injector that is created as a result of AdminResourceContainer startup routine.

2. AdminResourceContainer defined as a singleton has an instance of AdminPageRegistry that can be accessed with a getter. 

3. AdminResourceContainer allows setting a listenport explicitly. This along with #2 allows for potentially running multiple instances of AdminResourceContainers within a single JVM with a separate AdminPageRegistry associated with each instance.

### CODE REVIEW

**Comment by tbak** [comment]
This change still assumes that pages are discovered by classpath scanning. Can we have more control over how AdminResourceContainer is initialized?

**Comment by amit-git** [comment]
I see - so that's the default mode. I agree that it will be useful to disable classpath scanning (especially for unit tests, multiple admin instance use case) as a part of init config. Let me send another update on this PR

File: karyon2-admin/src/main/java/netflix/adminresources/AdminResourcesContainer.java [hunk 1]
```diff
@@ -144,15 +141,25 @@ public void init() throws Exception {
...
     public int getServerPort() {
-        return listenPort;
+        return serverPort;
+    }
+
+    public AdminPageRegistry getAdminPageRegistry() {
+        return adminPageRegistry;
     }
 
     private Module getAdditionalBindings() {
```
Comment (tbak) [inline karyon2-admin/src/main/java/netflix/adminresources/AdminResourcesContainer.java]: This method enforces AdminConfigImpl binding, while it should be possible to inject custom instance of it. Now injecting custom implementation results in duplicate binding exception.

**Comment by amit-git** [comment]
Verified with tbak that the API works as expected in case of unit test or multiple-instance setup. Merging.

**Pull request merged**